cluster/proxy_service.py

```
@@ -0,0 +1,267 @@
+        # Validate fingerprint ID is 32-bit unsigned integer
+        is_valid, fp_id = self._validate_fingerprint_id(fp_id_str)
+        if not is_valid:
+            return web.json_response({"error": "Fingerprint ID must be a 32-bit unsigned integer"}, status=400)
+        
+        # Delete publishes empty message using hex-encoded ID
+        fp_id_hex = f"{fp_id:08x}"
+        await self._publish_to_nats(index_name, fp_id_hex, "delete", {})
+        return web.json_response({"status": "deleted", "id": fp_id})
+    
+    # Bulk operations
+    async def bulk_update(self, request: web.Request) -> web.Response:
```

Ruff (0.12.2)

`bulk_update` is too complex (11 > 10)

(C901)